diff -up gdbm-1.8.0/gdbmopen.c.gdbmopen gdbm-1.8.0/gdbmopen.c
--- gdbm-1.8.0/gdbmopen.c.gdbmopen	2015-01-09 16:11:45.660400014 +0100
+++ gdbm-1.8.0/gdbmopen.c	2015-01-09 16:11:45.667400018 +0100
@@ -129,7 +129,7 @@ gdbm_open (file, block_size, flags, mode
 	dbf->desc = open (dbf->name, O_RDONLY, 0);
 	break;
 
-      case GDBM_OPENMASK:
+      case GDBM_WRITER:
 	dbf->desc = open (dbf->name, O_RDWR, 0);
 	break;
 
